### PR TITLE
QA-260 add check for tests without spec

### PR DESCRIPTION
**Pull request by annademina**

### CODE REVIEW

File: src/pytest_allure_spec_coverage/matcher.py [hunk 1]
```diff
@@ -210,19 +211,23 @@ def spec_coverage_percent(self):
...
+            test_ids = list(scenario_ids(item))
+            if not test_ids or any(test_id for test_id in test_ids if test_id not in sc_lookup):
+                self.nonexistent.append(item)
+            else:
+                for key in scenario_ids(item):
+                    with suppress(KeyError):
+                        if deselected:
+                            self.matches[sc_lookup[key]].deselected.append(item)
+                        else:
+                            self.matches[sc_lookup[key]].selected.append(item)
```
Comment (madlabman) [inline src/pytest_allure_spec_coverage/matcher.py:230]: What if to rewrite `suppress(KeyError)` expression, e.g.:

```suggestion
              for key in scenario_ids(item):
                  if key not in sc_lookup:
                      self.nonexistent.append(item)
                  else:
                      if deselected:
                          self.matches[sc_lookup[key]].deselected.append(item)
                      else:
                          self.matches[sc_lookup[key]].selected.append(item)
```
I see that I've removed a logic branch where test item is not marked at all, but I think that it is exactly what we want to - to collect non-existing scenarios ids.

Comment (annademina) [reply src/pytest_allure_spec_coverage/matcher.py]: thanks, changed!

File: tests/plugin/test_matcher.py [hunk 2]
```diff
@@ -231,13 +239,17 @@ def test_sc_only(pytester: Pytester):
...
+        ), 'Should be "50% specification coverage" in outlines'
     with allure.step("Assert that --sc-target less than coverage"):
         pytester_result, _ = run_with_allure(
             pytester=pytester,
             testfile_path="sc_only_test.py",
             additional_opts=["--sc-type", "test", "--sc-only", "--sc-target", "25"],
             outcomes={"passed": 0},
         )
-        assert "🎉🎉🎉" in pytester_result.outlines[-1]
+        assert any("🎉🎉🎉" in outline for outline in pytester_result.outlines), 'Should be "🎉🎉🎉" in outlines'
```
Comment (dgusakov) [inline tests/plugin/test_matcher.py:255]: Why `any`? there is only one condition here

File: src/pytest_allure_spec_coverage/matcher.py [hunk 3]
```diff
@@ -368,6 +370,9 @@ def pytest_terminal_summary(self, terminalreporter: TerminalReporter):
     def pytest_sessionfinish(self, session, exitstatus):
         """After terminal summary we print spec coverage success message"""
         yield
-        if self.config.fail_under and exitstatus != ExitCode.NO_TESTS_COLLECTED:
+        if (self.config.fail_under and exitstatus != ExitCode.NO_TESTS_COLLECTED) or self.nonexistent:
             terminal = session.config.pluginmanager.get_plugin("terminalreporter")
             terminal.write_line(f"Spec coverage is greater than target {self.config.fail_under}%! 🎉🎉🎉")
+            if self.nonexistent:
+                tests_without_spec = ", ".join([item.name for item in self.nonexistent])
+                terminal.write_line(f"There are tests without spec: {tests_without_spec}")
```
Comment (dgusakov) [inline src/pytest_allure_spec_coverage/matcher.py:378]: If there are a lot of such tests output will be messy. Consider adding list formatting

**Review by dgusakov — APPROVED** [review]

**Pull request merged**